### PR TITLE
feat(cli): Add new `info` command to output information about local environment

### LINKED ISSUE #2097: Having a dedicated info command to print out the local environment information
I think it is good to have a dedicated `info` command which will print out the local environment information that proves to be useful while submitting bug reports.
## Proposed API
`lerna info`

```bash
Environment Info:

  System:
    OS: Linux 4.18 Ubuntu 18.10 (Cosmic Cuttlefish)
    CPU: (2) x64 Intel(R) Pentium(R) CPU G2020 @ 2.90GHz
  Binaries:
    Node: 8.11.4 - /usr/bin/node
    Yarn: 1.13.0 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 72.0.3626.81
    Firefox: 65.0
```

I would be happy to work on this :clap:

### PR DESCRIPTION
**Pull request by jamesgeorge007**
## Description
Added `lerna info` command which prints local environment information.

## Motivation and Context

Closes #2097 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: commands/info/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+module.exports = factory;
+
+function factory(argv) {
+  return new InfoCommand(argv);
+}
+
+class InfoCommand extends Command {
+  execute() {
+    output("\n Environment info:");
```
Comment (evocateur) [inline commands/info/index.js]: This strikes me as kinda redundant. I think we only need the actual output of `envinfo.run()`

File: commands/info/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    const info = await envinfo.run({
```
Comment (evocateur) [inline commands/info/index.js]: Lerna 3.x still supports node v6.x, so we can't use `async`/`await` yet. Simple enough to refactor:
```js
return Promise.resolve()
  .then(() =>
    envinfo.run({
      // config
    })
  })
  .then(output);
```

File: commands/info/__tests__/info-command.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+// mocked modules
+// const output = require("@lerna/output");
```
[the remark this answers is not in the document]
Comment (jamesgeorge007) [reply commands/info/__tests__/info-command.test.js]: @evocateur Can you elaborate?

File: commands/info/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+const envinfo = require("envinfo");
+
+module.exports = factory;
+
+function factory(argv) {
+  return new InfoCommand(argv);
+}
+
+class InfoCommand extends Command {
+  execute() {
```
[the remark this answers is not in the document]
Comment (evocateur) [reply commands/info/index.js]: I still don't see a no-op `initialize() {}` method

File: commands/info/index.js (hunk 2 above)
Comment (evocateur) [reply commands/info/index.js]: This isn't resolved, it's still using `await`

File: commands/info/__tests__/info-command.test.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (evocateur) [reply commands/info/__tests__/info-command.test.js]: At the top of the file:
```js
jest.mock("envinfo");

const envinfo = require("envinfo");

envinfo.run.mockResolvedValue("MOCK_ENVINFO");
```

In the test:
```js
  it("outputs result of envinfo()", async () => {
    // project fixture is irrelevant, no actual changes are made
    await lernaInfo(process.cwd())();

    expect(envinfo.run).toHaveBeenLastCalledWith(
      expect.objectContaining({
        npmPackages: ["lerna"],
      })
    );
    expect(output.logged()).toMatch("MOCK_ENVINFO");
  });
```

File: core/lerna/package.json [hunk 5]
```diff
@@ -44,6 +44,7 @@
     "@lerna/diff": "file:../../commands/diff",
     "@lerna/exec": "file:../../commands/exec",
     "@lerna/import": "file:../../commands/import",
+    "@lerna/info": "file:../../commands/info",
```
Comment (evocateur) [inline core/lerna/package.json]: You'll need to run `npm i` in the root of the repo, as well. There should be a new entry in `package-lock.json`

**Comment by jamesgeorge007** [comment]
cc @evocateur

File: package-lock.json [hunk 6]
```diff
@@ -2909,6 +2917,10 @@
         "once": "^1.4.0"
       }
     },
+    "envinfo": {
+      "version": "7.3.1",
+      "resolved": "https://registry.npmjs.org/envinfo/-/envinfo-7.3.1.tgz",
+      "integrity": "sha512-GvXiDTqLYrORVSCuJCsWHPXF5BFvoWMQA9xX4YVjPT1jyS3aZEHUBwjzxU/6LTPF9ReHgVEbX7IEN5UvSXHw/A=="
```
Comment (evocateur) [inline package-lock.json]: Something got mangled here, the object should close.

Comment (jamesgeorge007) [reply package-lock.json]: Oops!

**Comment by jamesgeorge007** [comment]
cc @evocateur

File: commands/info/__tests__/info-command.test.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+envinfo.run.mockResolvedValue("MOCK_ENVINFO");
+
+// helper
+const output = require("@lerna/output");
+
+// file under test
+const lernaInfo = require("@lerna-test/command-runner")(require("../command"));
+
+it("outputs result of envinfo()", async () => {
+  // project fixture is irrelevant, no actual changes are made
```
Comment (evocateur) [inline commands/info/__tests__/info-command.test.js]: Actually, it does matter, as evidenced by the test failures. You can use the `lifecycle` fixture, it's small enough.

Comment (evocateur) [reply commands/info/__tests__/info-command.test.js]: (That, or the `process.cwd()` below needs to be a more explicit path, like `path.resolve(__dirname, '../../..')`)

**Pull request merged**